File: UI/rentify_desktop/lib/models/payment.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:rentify_desktop/models/reservation.dart';
import 'package:rentify_desktop/models/status.dart';
import 'package:rentify_desktop/models/user.dart';

part 'payment.g.dart';

@JsonSerializable()
class Payment {
  final int id;
  final int reservationId;
  Reservation? reservation;
  final String name;
  final String comment;
  final double price;
  final int monthNumber;
  final int yearNumber;
  final int statusId;
  Status? status;
  DateTime? dateToPay;
  DateTime? warningDateToPay;
  DateTime? secondWarningDate;

  Payment({
    required this.id,
    required this.reservationId,
    this.reservation,
    required this.name,
    required this.comment,
    required this.price,
    required this.monthNumber,
    required this.yearNumber,
    required this.statusId,
    this.status,
    this.dateToPay,
    this.warningDateToPay,
    this.secondWarningDate,
  });

  factory Payment.fromJson(Map<String, dynamic> json) =>
      _$PaymentFromJson(json);

  String? get paymentStatus => status?.name;
  String? get reservationStatus => reservation?.status?.name;

  Map<String, dynamic> toJson() => _$PaymentToJson(this);
}

File: Rentify.Services/Database/Payment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Rentify.Services.Database
{
    public class Payment
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey(nameof(ReservationId))]
        public int ReservationId { get; set; }
        public Reservation? Reservation { get; set; }

        public string Name { get; set; }
        public string Comment { get; set; }
        public decimal Price { get; set; }
        public int MonthNumber { get; set; }
        public int YearNumber { get; set; }
        public DateTime? DateToPay { get; set; }
        public DateTime? WarningDateToPay { get; set; }
        public DateTime? SecondWarningDate { get; set; }
        public string? StripePaymentIntentId { get; set; }
        public DateTime? PaidAt { get; set; }

        [ForeignKey(nameof(StatusId))]
        public int StatusId { get; set; } = 1;
        public Status? Status { get; set; }
    }
}
